### PR TITLE
[Editor] Handle correctly colors when saving a document in HCM

### PR DESCRIPTION
**Pull request by calixteman**
- for example in Dusk theme (Windows 11), black appears to be white, so
  the user will draw something in white. But if they want to print or
  save the used color must be black.
- fix a bug with the color input which only accepts hex string colors;
- adjust outline color of the selected/hovered editors in HCM.

### CODE REVIEW

File: src/display/editor/tools.js [hunk 1]
```diff
@@ -279,6 +280,63 @@ class ClipboardManager {
...
 }
 
+class ColorManager {
+  static _colorsMapping = new Map([
+    ["CanvasText", [0, 0, 0]],
+    ["Canvas", [255, 255, 255]],
+  ]);
+
+  get _colors() {
+    if (typeof document !== "undefined") {
```
Comment (Snuffleupagus) [inline src/display/editor/tools.js:290]: Why do we need this check, since I don't believe this code is expected to run outside of browsers?
Also, in the method just below you're accessing `window` *directly* without checking if it's available first.

File: src/display/editor/tools.js [hunk 2]
```diff
@@ -279,6 +280,63 @@ class ClipboardManager {
...
+   * So this function converts a name into an hex string.
+   * @param {string} name
+   * @returns {string}
+   */
+  getHexCode(name) {
+    const rgb = this._colors.get(name);
+    if (!rgb) {
+      return name;
+    }
+    return "#" + rgb.map(x => x.toString(16).padStart(2, "0")).join("");
```
Comment (Snuffleupagus) [inline src/display/editor/tools.js:336]: Wait, isn't this *identical* to the existing `Util.makeHexColor` method?
Please use that one instead, rather than reinventing the wheel here :-)

File: src/display/display_utils.js [hunk 3]
```diff
@@ -589,12 +589,25 @@ function getRGB(color) {
   return [0, 0, 0];
 }
 
+function getColorValues(colors) {
```
Comment (Snuffleupagus) [inline src/display/display_utils.js:592]: Are you expecting to use this anywhere *except* in the Editor-layer?
If not, should we perhaps just place this code in the `src/display/editor/tools.js` file instead?

File: src/display/editor/tools.js (hunk 1 above)
Comment (calixteman) [reply src/display/editor/tools.js:290]: I added this check to fix `gulp unittestcli` which was unhappy with that (likely because a static property is using it)... so I chose the easiest solution.

File: src/display/editor/tools.js (hunk 2 above)
Comment (calixteman) [reply src/display/editor/tools.js:336]: Especially when I created this wheel myself...

File: src/display/display_utils.js (hunk 3 above)
Comment (calixteman) [reply src/display/display_utils.js:592]: I thought it could be reused to get the real colors we pass to the canvas in HCM.
We do something similar here:
https://github.com/mozilla/pdf.js/blob/ce63663282e9f7fff4adecf11cc030f436d7d4cc/src/display/canvas.js#L1291-L1292

File: src/display/editor/tools.js (hunk 1 above)
Comment (Snuffleupagus) [reply src/display/editor/tools.js:290]: Ah, that explains things. In that case, since this doesn't matter in browser, please see the suggestion below (which based on what we do in the viewer-code in a couple of spots).

File: src/pdf.js [hunk 4]
```diff
@@ -123,6 +124,7 @@ export {
   getDocument,
   getFilenameFromUrl,
   getPdfFilenameFromUrl,
+  getRGB,
```
Comment (Snuffleupagus) [inline src/pdf.js:127]: Why is this necessary, since I'm not finding anything that depends on this?
As a general rule, please let's not expose things *in the API* unless it's actually needed e.g. in the viewer.

**Review by Snuffleupagus — APPROVED** [review]
r=me, with comments addressed and passing tests; thank you!

**Comment by calixteman** [comment]
So it's my fault:
image

**Comment by calixteman** [comment]
To fix the issue, I lazily get the static values computed thanks to `document`:
https://github.com/mozilla/pdf.js/pull/15115/files#diff-8ba8badc29ff13ad2d4b61ad2b7df503933c69c028150d1ba24188deb7f11493R61-R68
